common/lib/session_state_service_impl.ts

```diff
@@ -57,68 +58,13 @@ export class SessionStateServiceImpl implements SessionStateService {
       }
     }
 
-    if (this.sessionState.autoCommit.value !== undefined) {
-      this.sessionState.autoCommit.resetPristineValue();
-      this.setupPristineAutoCommit();
-      try {
-        await newClient.setAutoCommit(this.sessionState.autoCommit.value);
-      } catch (error: any) {
-        if (error instanceof UnsupportedMethodError) {
-          // ignore
-        }
-        throw error;
-      }
-    }
-
-    if (this.sessionState.readOnly.value !== undefined) {
-      this.sessionState.readOnly.resetPristineValue();
-      this.setupPristineReadOnly();
-      try {
-        await newClient.updateSessionStateReadOnly(this.sessionState.readOnly.value);
-      } catch (error: any) {
-        if (error instanceof UnsupportedMethodError) {
-          // ignore
-        }
-        throw error;
-      }
-    }
+    const targetClient: ClientWrapper = newClient.targetClient;
 
-    if (this.sessionState.catalog.value !== undefined) {
-      this.sessionState.catalog.resetPristineValue();
-      this.setupPristineCatalog();
-      try {
-        await newClient.setCatalog(this.sessionState.catalog.value);
-      } catch (error: any) {
-        if (error instanceof UnsupportedMethodError) {
-          // ignore
-        }
-        throw error;
-      }
-    }
-
-    if (this.sessionState.schema.value !== undefined) {
-      this.sessionState.schema.resetPristineValue();
-      this.setupPristineSchema();
-      try {
-        await newClient.setSchema(this.sessionState.schema.value);
-      } catch (error: any) {
-        if (error instanceof UnsupportedMethodError) {
-          // ignore
-        }
-        throw error;
-      }
-    }
-
-    if (this.sessionState.transactionIsolation.value !== undefined) {
-      this.sessionState.transactionIsolation.resetPristineValue();
-      this.setupPristineTransactionIsolation();
-      try {
-        await newClient.setTransactionIsolation(this.sessionState.transactionIsolation.value);
-      } catch (error: any) {
-        if (error instanceof UnsupportedMethodError) {
-          // ignore
-        }
-        throw error;
+    // Apply current state for all 5 states: autoCommit, readOnly, catalog, schema, transactionIsolation
+    for (const key of Object.keys(this.sessionState)) {
+      const state = this.sessionState[key];
+      if (state instanceof SessionStateField) {
```

Comment:
We need "else" with an exception thrown. If the value isn't what we expected it's an internal error.